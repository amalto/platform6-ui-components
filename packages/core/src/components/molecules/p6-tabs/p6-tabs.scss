@import '@fortawesome/fontawesome-svg-core/styles';
@import '@creativebulma/bulma-tooltip/src/sass/index';

$tabs-link-active-border-top-color: $primary;
$tabs-link-active-color: $success;

$tabs-border-bottom-color: $border !default;
$tabs-border-bottom-style: solid !default;
$tabs-border-bottom-width: 1px !default;
$tabs-link-color: $text !default;
$tabs-link-hover-border-bottom-color: $text-strong !default;
$tabs-link-hover-color: $text-strong !default;
$tabs-link-active-border-top-color: $link !default;
$tabs-link-active-color: $link !default;
$tabs-link-padding: 0.5em 1em !default;
$tabs-link-max-width: 7rem;

$tabs-boxed-link-radius: $radius !default;
$tabs-boxed-link-hover-background-color: $background !default;
$tabs-boxed-link-hover-border-bottom-color: $border !default;

$tabs-boxed-link-active-background-color: $scheme-main !default;
$tabs-boxed-link-active-border-color: $border !default;
$tabs-boxed-link-active-border-bottom-color: transparent !default;

$tabs-toggle-link-border-color: $border !default;
$tabs-toggle-link-border-style: solid !default;
$tabs-toggle-link-border-width: 1px !default;
$tabs-toggle-link-hover-background-color: $background !default;
$tabs-toggle-link-hover-border-color: $border-hover !default;
$tabs-toggle-link-radius: $radius !default;
$tabs-toggle-link-active-background-color: $link !default;
$tabs-toggle-link-active-border-color: $link !default;
$tabs-toggle-link-active-color: $link-invert !default;

:host {
  display: block;

  .tabs {
    -webkit-overflow-scrolling: touch;
    @include unselectable;

    align-items: stretch;
    display: flex;
    font-size: $size-normal;
    justify-content: space-between;
    overflow: visible;
    white-space: nowrap;
    margin-bottom: 0;
    position: relative;
    z-index: 100;

    a {
      align-items: center;
      border-bottom-color: $tabs-border-bottom-color;
      border-bottom-style: $tabs-border-bottom-style;
      border-bottom-width: $tabs-border-bottom-width;
      color: $tabs-link-color;
      display: flex;
      justify-content: center;
      margin-bottom: -#{$tabs-border-bottom-width};
      padding: $tabs-link-padding;
      vertical-align: top;
      cursor: pointer;
      text-decoration: none !important;
      position: relative;
      max-width: $tabs-link-max-width;

      .title {
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
      }

      .delete.disabled-close {
        display: none;
      }

      .delete:not(.disabled-close) {
        @include delete;
      }

      &:hover {
        border-bottom-color: $tabs-link-hover-border-bottom-color;
        color: $tabs-link-hover-color;
      }
    }

    &.is-boxed a {
      border: 1px solid transparent;
      border-radius: $tabs-boxed-link-radius $tabs-boxed-link-radius 0 0;
    }
    &.is-toggle a {
      border-color: $tabs-toggle-link-border-color;
      border-style: $tabs-toggle-link-border-style;
      border-width: $tabs-toggle-link-border-width;
      margin-bottom: 0;
      position: relative;
    }

    li {
      display: block;

      a:not(.is-active) {
        background-color: $grey-lightest;
        border-left: 1px solid $white;
        border-right: 1px solid $white;
      }

      a.is-active,
      &.is-active a {
        border-top: 2px solid $tabs-link-active-border-top-color;
        border-left: 1px solid $tabs-boxed-link-hover-border-bottom-color;
        border-right: 1px solid $tabs-boxed-link-hover-border-bottom-color;
        border-bottom-color: $white;
        color: $tabs-link-active-color;
      }
    }

    &.is-boxed a:hover {
      background-color: $tabs-boxed-link-hover-background-color;
      border-bottom-color: $tabs-boxed-link-hover-border-bottom-color;
    }

    &.is-toggle a:hover {
      background-color: $tabs-toggle-link-hover-background-color;
      border-color: $tabs-toggle-link-hover-border-color;
      z-index: 2;
    }

    li a:hover:not(.is-active) {
      border-bottom-color: $tabs-link-hover-border-bottom-color;
      color: $tabs-link-hover-color;
    }

    ul {
      align-items: center;
      border-bottom-color: $tabs-border-bottom-color;
      border-bottom-style: $tabs-border-bottom-style;
      border-bottom-width: $tabs-border-bottom-width;
      display: flex;
      flex-grow: 1;
      flex-shrink: 0;
      justify-content: flex-start;
      margin: 0;
      padding: 0;

      &.is-left {
        padding-right: 0.75em;
      }

      &.is-center {
        flex: none;
        justify-content: center;
        padding-left: 0.75em;
        padding-right: 0.75em;
      }

      &.is-right {
        justify-content: flex-end;
        padding-left: 0.75em;
      }
    }

    .icon {
      &:first-child {
        margin-right: 0.5em;
      }

      &:last-child {
        margin-left: 0.5em;
      }
    }

    // Alignment
    &.is-centered {
      ul {
        justify-content: center;
      }
    }

    &.is-right {
      ul {
        justify-content: flex-end;
      }
    }

    // Styles
    &.is-boxed li {
      &.is-active {
        a {
          background-color: $tabs-boxed-link-active-background-color;
          border-color: $tabs-boxed-link-active-border-color;
          border-bottom-color: $tabs-boxed-link-active-border-bottom-color !important;
        }
      }
    }

    &.is-fullwidth {
      li {
        flex-grow: 1;
        flex-shrink: 0;
      }
    }

    &.is-toggle {
      li {
        & + li {
          margin-left: -#{$tabs-toggle-link-border-width};
        }

        &:first-child a {
          border-top-left-radius: $tabs-toggle-link-radius;
          border-bottom-left-radius: $tabs-toggle-link-radius;
        }

        &:last-child a {
          border-top-right-radius: $tabs-toggle-link-radius;
          border-bottom-right-radius: $tabs-toggle-link-radius;
        }

        &.is-active {
          a {
            background-color: $tabs-toggle-link-active-background-color;
            border-color: $tabs-toggle-link-active-border-color;
            color: $tabs-toggle-link-active-color;
            z-index: 1;
          }
        }
      }

      ul {
        border-bottom: none;
      }

      &.is-toggle-rounded {
        li {
          &:first-child a {
            border-bottom-left-radius: $radius-rounded;
            border-top-left-radius: $radius-rounded;
            padding-left: 1.25em;
          }

          &:last-child a {
            border-bottom-right-radius: $radius-rounded;
            border-top-right-radius: $radius-rounded;
            padding-right: 1.25em;
          }
        }
      }
    }

    // Sizes
    &.is-small {
      font-size: $size-small;
    }

    &.is-medium {
      font-size: $size-medium;
    }

    &.is-large {
      font-size: $size-large;
    }
  }

  .tab-content {
    padding: $gap-size * 2 * 1px;
    border-left: 1px solid $tabs-border-bottom-color;
    border-right: 1px solid $tabs-border-bottom-color;
    border-bottom: 1px solid $tabs-border-bottom-color;
  }
}
