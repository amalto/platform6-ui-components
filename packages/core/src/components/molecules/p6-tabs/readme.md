# p6-tabs

<!-- Auto Generated Below -->

## Events

| Event        | Description     | Type                              |
| ------------ | --------------- | --------------------------------- |
| `p6CloseTab` | Close tab event | `CustomEvent<{ tabId: string; }>` |

## Methods

### `close(tabId: string) => Promise<boolean>`

close a tab

#### Returns

Type: `Promise<boolean>`

### `refresh() => Promise<void>`

refresh the component

#### Returns

Type: `Promise<void>`

---

_Built with [StencilJS](https://stenciljs.com/)_
