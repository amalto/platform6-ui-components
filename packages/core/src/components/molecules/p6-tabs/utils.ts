export function getTabId(id: string): string {
  return `${id}-tab`;
}

export function isTabValid(tab: Element): tab is HTMLElement {
  return 'title' in tab && 'id' in tab;
}
