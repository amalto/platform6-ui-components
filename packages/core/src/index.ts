export { Components, JSX } from './components';
export { P6ButtonType } from './components/atoms/p6-button/p6-button';
export { P6CalendarType } from './components/atoms/p6-calendar/p6-calendar';
export { P6CheckboxValue } from './components/atoms/p6-checkbox/p6-checkbox';
export { P6InputValue } from './components/atoms/p6-input/p6-input';
export { P6InputType } from './components/atoms/p6-input/types';
export { Target } from './components/atoms/p6-link/p6-link';
export { P6RadioValue } from './components/atoms/p6-radio/p6-radio';
export { P6SelectControl, P6SelectValue } from './components/atoms/p6-select-native/p6-select-native';
export { P6SwitchValue } from './components/atoms/p6-switch/p6-switch';
export { P6TextareaValue } from './components/atoms/p6-textarea/p6-textarea';
export { P6LanguageValue } from './components/molecules/p6-language/p6-language';
export { P6TranslationControl, P6TranslationValue } from './components/molecules/p6-translation/p6-translation';
export {
  AlignColumnDetail,
  CellValueChangedDetail,
  Column,
  ColumnDefinition,
  ColumnId,
  DataItem,
  EditingCellDetail,
  FilterRowsDetail,
  MoveColumnDetail,
  ResetDefinitionsDetail,
  ResizeColumnDetail,
  Row,
  RowId,
  ShowColumnDetail,
  ShowOptionsDetail,
  SortColumnDetail,
} from './components/organisms/p6-grid/core/entities';
export { ServiceStatus } from './components/templates/p6-service-panel/service-status.enum';
export { P6Control } from './shared/form/control';
export { InvalidEventDetail, isCustomEvent, isInvalidEvent, isValidEvent, ValidEventDetail } from './shared/form/event';
export { Alignment, Mode, modes, Position, positions, Size, sizes, SortOrder } from './shared/types';
export { EnumArrayEntry, enumArrayToObject, enumToArray, EnumType } from './utils/enum';
export { LanguageCode } from './utils/language';
