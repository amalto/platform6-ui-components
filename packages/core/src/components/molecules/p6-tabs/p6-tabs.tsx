import { Component, ComponentInterface, Element, Event, EventEmitter, forceUpdate, h, Host, JSX, Method, State } from '@stencil/core';
import { toArray } from '../../../utils/dom';
import { isTabValid } from './utils';

@Component({
  tag: 'p6-tabs',
  styleUrl: 'p6-tabs.scss',
  shadow: true,
})
export class P6Tabs implements ComponentInterface {
  @Element() host!: HTMLP6TabsElement;

  /**
   * Close tab event
   */
  @Event({ eventName: 'p6CloseTab' }) closeTabEmitter?: EventEmitter<{ tabId: string }>;

  @State() selectedTabId: string | undefined;

  /**
   * refresh the component
   */
  @Method()
  async refresh(): Promise<void> {
    forceUpdate(this.host);
  }

  /**
   * close a tab
   * @param tabId the id of the tab to be close
   */
  @Method()
  async close(tabId: string): Promise<boolean> {
    const tabToClose = this.host.querySelector<HTMLP6TabElement>(`#${tabId}`);

    if (tabToClose !== null) {
      this.closeTab(tabToClose);
      return Promise.resolve(true);
    }
    return Promise.resolve(false);
  }

  private getTabs(): HTMLP6TabElement[] {
    return toArray(this.host.children).filter(isTabValid);
  }

  componentWillLoad(): void {
    const tabs = this.getTabs();
    const selectedTab = tabs.find(tab => tab.active);

    if (tabs.length === 0) {
      return;
    }

    if (selectedTab === undefined) {
      tabs[0].active = true;
      this.selectedTabId = tabs[0].id;
    } else {
      this.selectedTabId = selectedTab.id;
    }
  }

  componentWillRender(): void {
    this.getTabs().forEach(tab => {
      if (tab.closed) {
        this.host.removeChild(tab);
      }
    });
  }

  render(): JSX.Element | null {
    const tabs = this.getTabs();

    if (tabs.length === 0) {
      return null;
    }

    return (
      <Host>
        <div class="tabs">
          <ul>{tabs.map(tab => this.renderTab(tab))}</ul>
        </div>
        <div class="tab-content">
          <slot />
        </div>
      </Host>
    );
  }

  private renderTab(tab: HTMLP6TabElement): JSX.Element {
    return (
      <li class="has-tooltip-arrow has-tooltip-bottom" data-tooltip={tab.title}>
        <a class={tab.active ? 'is-active' : undefined} href={`#${tab.id}`} onClick={this.selectTabHandler(tab)}>
          <span class="title">{tab.title}</span>
          <span aria-hidden="true" class={{ 'delete': true, 'disabled-close': !tab.closeable }} onClick={this.closeTabHandler(tab)} />
        </a>
      </li>
    );
  }

  private selectTabHandler(selectedTab: HTMLP6TabElement): (event: MouseEvent) => void {
    return (event: MouseEvent): void => {
      event.preventDefault();

      this.getTabs().forEach(tab => {
        // eslint-disable-next-line no-param-reassign
        tab.active = tab.id === selectedTab.id;
      });
      this.selectedTabId = selectedTab.id;
    };
  }

  private closeTabHandler(closedTab: HTMLP6TabElement): (event: MouseEvent) => void {
    return (event: MouseEvent): void => {
      event.preventDefault();
      event.stopPropagation();

      this.closeTab(closedTab);
    };
  }

  private closeTab(tabToclose: HTMLP6TabElement): void {
    tabToclose.close();

    const tabs = this.getTabs();
    const closedTabIndex = tabs.findIndex(tab => tab.id === tabToclose.id);

    if (tabToclose.active && tabs[closedTabIndex - 1] !== undefined) {
      tabs[closedTabIndex - 1].active = true;
      this.selectedTabId = tabs[closedTabIndex - 1].id;
    } else {
      forceUpdate(this.host);
    }

    this.closeTabEmitter?.emit({ tabId: tabToclose.id });
  }
}
