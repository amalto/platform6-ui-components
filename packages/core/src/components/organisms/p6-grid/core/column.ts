import { Alignment, Direction, SortOrder } from '../../../../shared/types';
import { throwError } from '../../../../utils/error';
import { Column, ColumnDefinition, ColumnId, DataItem } from './entities';

export const MIN_WIDTH = 75;
export const DEFAULT_WIDTH = MIN_WIDTH;
export const INC_WIDTH = 10;

export function defaultGetValue<T extends DataItem>(data: T, context: ColumnDefinition<T>): string {
  if (context.field === undefined || data[context.field] === undefined) {
    return throwError(`field is undefined or doesn't exist, you should provide a 'getValue' function for ${JSON.stringify(context)}`);
  }

  return `${data[context.field as keyof T]}`;
}

export function defaultSetValue<T extends DataItem>(value: string, data: T, context: ColumnDefinition<T>): T {
  if (context.field === undefined || data[context.field] === undefined) {
    return throwError(`field is undefined or doesn't exist, you should provide a 'setValue' function for ${JSON.stringify(context)}`);
  }

  return {
    ...data,
    [context.field as keyof T]: value,
  };
}

let columnIds = 0;
export function fromDefinition<T extends DataItem>(column: ColumnDefinition<T>): Column<T> {
  // eslint-disable-next-line no-plusplus
  const id = column.id || column.field || columnIds++;

  return {
    ...column,
    id: `col-${id}`,
    editable: column.editable || false,
    hidden: column.hidden || false,
    width: column.width || 100,
    align: column.align || Alignment.center,
    sortOrder: column.sortOrder || SortOrder.none,
    disableHeaderMenu: column.disableHeaderMenu || false,
    getValue: column.getValue || defaultGetValue,
    setValue: column.setValue || defaultSetValue,
  };
}

export function isHidden<T extends DataItem>(column: Column<T>): boolean {
  return !!column.hidden;
}

export function replaceColumn<T extends DataItem>(updatedColumn: Column<T>, allColumns: Column<T>[]): Column<T>[] {
  return allColumns.map(col => (col.id === updatedColumn.id ? updatedColumn : col));
}

export function move<T extends DataItem>(id: ColumnId, direction: Direction, columns: Column<T>[]): Column<T>[] {
  const from = columns.findIndex(col => col.id === id);
  const to = from + (direction === Direction.left ? -1 : 1);

  if (from < 0 || to < 0 || to >= columns.length) {
    return columns;
  }

  const result = columns.slice();
  const item = result.splice(from, 1);

  return result.slice(0, to).concat(item).concat(result.slice(to, columns.length));
}
