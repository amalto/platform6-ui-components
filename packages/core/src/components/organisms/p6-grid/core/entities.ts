import { EventEmitter, JSX } from '@stencil/core';
import { Alignment, Direction, SortOrder } from '../../../../shared/types';

export type ColumnId = string;

export type DataItem = Record<string, unknown>;

export interface ColumnDefinition<T extends DataItem> {
  id?: ColumnId;
  label: string;
  field?: string | keyof T;
  editable?: boolean;
  filtreable?: boolean;
  sortable?: boolean;
  hidden?: boolean;
  width?: number;
  align?: Alignment;
  color?: string;
  sortOrder?: SortOrder;
  disableHeaderMenu?: boolean;
  getValue?: (data: T, context: ColumnDefinition<T>) => string;
  setValue?: (value: string, data: T, context: ColumnDefinition<T>) => T;
  cellEditor?: (data: T, context: ColumnDefinition<T>) => JSX.Element & CellEditor<T>;
  cellRenderer?: (data: T, rowId: RowId, context: ColumnDefinition<T>) => string | HTMLElement;
}

export interface Column<T extends DataItem> extends ColumnDefinition<T> {
  id: ColumnId;
  editable: boolean;
  hidden: boolean;
  width: number;
  align: Alignment;
  sortOrder: SortOrder;
  disableHeaderMenu: boolean;
  getValue: (data: T, context: ColumnDefinition<T>) => string;
  setValue: (value: string, data: T, context: ColumnDefinition<T>) => T;
}

export interface CellEditor<T extends DataItem> {
  column: Column<T>;
  row: Row<T>;
  value: string;

  p6GridCellValueChanged: EventEmitter<CellValueChangedDetail<T>>;
  p6GridEditingCell: EventEmitter<EditingCellDetail<T>>;
}
export type RowId = string;

export interface Row<T extends DataItem> {
  id: RowId;
  data: T;
}

interface ColumnEventDetail<T extends DataItem> {
  column: Column<T>;
}

export interface MoveColumnDetail<T extends DataItem> extends ColumnEventDetail<T> {
  direction: Direction;
}
export type ResizeColumnDetail<T extends DataItem> = ColumnEventDetail<T>;

export type AlignColumnDetail<T extends DataItem> = ColumnEventDetail<T>;

export type SortColumnDetail<T extends DataItem> = ColumnEventDetail<T>;

export type ShowColumnDetail<T extends DataItem> = ColumnEventDetail<T>;

export interface CellValueChangedDetail<T extends DataItem> {
  column: Column<T>;
  row: Row<T>;
  oldValue: string;
}

export enum EditingCellStatus {
  Start,
  Stop,
  Cancel,
}

export interface EditingCellDetail<T extends DataItem> {
  column: Column<T>;
  row: Row<T>;
  status: EditingCellStatus;
}

export interface ResetDefinitionsDetail {
  reset: boolean;
}

export interface ShowOptionsDetail {
  visible: boolean;
}

export interface FilterRowsDetail {
  value: string;
}
