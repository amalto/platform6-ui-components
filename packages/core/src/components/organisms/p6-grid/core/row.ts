import { SortOrder } from '../../../../shared/types';
import { Column, DataItem, Row, RowId } from './entities';

let rowIds = 0;
export function fromData<T extends DataItem>(data: T): Row<T> {
  rowIds += 1;

  return {
    id: `row-${rowIds}`,
    data,
  };
}

export function compareRow<T extends DataItem>(column: Column<T> | undefined): (a: Row<T>, b: Row<T>) => number {
  const sortOrder = column?.sortOrder || SortOrder.none;
  const factor = sortOrder === SortOrder.desc ? -1 : 1;

  return (a: Row<T>, b: Row<T>): number => {
    if (column === undefined || column.sortable === false || sortOrder === SortOrder.none) {
      return 0;
    }

    const valueA = column.getValue(a.data, column);
    const valueB = column.getValue(b.data, column);

    return valueA.localeCompare(valueB) * factor;
  };
}

export function filterBy<T extends DataItem>(searchValue: string, columns: Column<T>[]): (row: Row<T>) => boolean {
  return row => columns.filter(col => col.filtreable !== false).some(col => col.getValue(row.data, col).includes(searchValue));
}

export function rangeSelectRow<T extends DataItem>(currentRowIdSelect: RowId, selectedRows: Set<RowId>, displayedRow: Row<T>[]): Set<RowId> {
  const currentSelectedIndex = displayedRow.findIndex(row => row.id === currentRowIdSelect);

  if (currentSelectedIndex === -1) {
    return selectedRows;
  }

  const previousRowIdSelected = Array.from(selectedRows)
    .reverse()
    .find(rowId => displayedRow.findIndex(row => row.id === rowId) > -1);

  if (previousRowIdSelected === undefined) {
    return new Set(selectedRows.add(currentRowIdSelect));
  }

  const previousSelectedIndex = displayedRow.findIndex(row => row.id === previousRowIdSelected);

  let [from, to] = [previousSelectedIndex, currentSelectedIndex];
  let excludedFromSlice = currentRowIdSelect;
  if (currentSelectedIndex < previousSelectedIndex) {
    [from, to] = [to, from];
    excludedFromSlice = previousRowIdSelected;
  }

  return displayedRow
    .slice(from, to)
    .reduce((selected, cur) => {
      return cur.id === undefined ? selected : selected.add(cur.id);
    }, new Set(selectedRows))
    .add(excludedFromSlice);
}

export function replaceRow<T extends DataItem>(updatedRow: Row<T>, allRows: Row<T>[]): Row<T>[] {
  return allRows.map(row => (row.id === updatedRow.id ? updatedRow : row));
}
