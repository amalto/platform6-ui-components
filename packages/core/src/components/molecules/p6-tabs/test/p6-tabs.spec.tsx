import { newSpecPage } from '@stencil/core/testing';
import { P6Tabs } from '../p6-tabs';

describe('p6-tabs', () => {
  it('Should be empty', async () => {
    const page = await newSpecPage({
      components: [P6Tabs],
      html: `<p6-tabs></p6-tabs>`,
    });

    expect(page.root).toEqualHtml(`
      <p6-tabs>
        <mock:shadow-root>
         </mock:shadow-root>
      </p6-tabs>
    `);
  });

  it('Should display the third tab', async () => {
    const page = await newSpecPage({
      components: [P6Tabs],
      html: `
      <p6-tabs selected="third">
        <p6-tab id="first" title="What is Lorem Ipsum?">1</p6-tab>
        <p6-tab id="second" title="Where does it come from?">2</p6-tab>
        <p6-tab id="third" title="Why do we use it?">3</p6-tab>
        <p6-tab id="fourth" title="Where can I get some?">4</p6-tab>
      </p6-tabs>
      `,
    });

    expect(page.root).toMatchSnapshot();
  });
});
