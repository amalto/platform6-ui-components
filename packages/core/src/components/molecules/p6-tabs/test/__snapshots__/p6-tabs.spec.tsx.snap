// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-tabs Should display the third tab 1`] = `
<p6-tabs selected="third">
  <mock:shadow-root>
    <div class="tabs">
      <ul>
        <li class="has-tooltip-arrow has-tooltip-bottom" data-tooltip="What is Lorem Ipsum?">
          <a class="is-active" href="#first">
            <span class="title">
              What is Lorem Ipsum?
            </span>
            <span aria-hidden="true" class="delete disabled-close"></span>
          </a>
        </li>
        <li class="has-tooltip-arrow has-tooltip-bottom" data-tooltip="Where does it come from?">
          <a href="#second">
            <span class="title">
              Where does it come from?
            </span>
            <span aria-hidden="true" class="delete disabled-close"></span>
          </a>
        </li>
        <li class="has-tooltip-arrow has-tooltip-bottom" data-tooltip="Why do we use it?">
          <a href="#third">
            <span class="title">
              Why do we use it?
            </span>
            <span aria-hidden="true" class="delete disabled-close"></span>
          </a>
        </li>
        <li class="has-tooltip-arrow has-tooltip-bottom" data-tooltip="Where can I get some?">
          <a href="#fourth">
            <span class="title">
              Where can I get some?
            </span>
            <span aria-hidden="true" class="delete disabled-close"></span>
          </a>
        </li>
      </ul>
    </div>
    <div class="tab-content">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <p6-tab id="first" title="What is Lorem Ipsum?">
    1
  </p6-tab>
  <p6-tab id="second" title="Where does it come from?">
    2
  </p6-tab>
  <p6-tab id="third" title="Why do we use it?">
    3
  </p6-tab>
  <p6-tab id="fourth" title="Where can I get some?">
    4
  </p6-tab>
</p6-tabs>
`;
